lib/clamp-link.in

```diff
@@ -119,6 +119,230 @@ _find_object() {
+    fi
+  fi
+done
+
+
+
+
+for ARG in $ARGS
+do
+
+  ####################################
+  #  parse the simple switches first...
+  ####################################
+
+  if [ $ARG == "--disable-opencl" ]; then
+    LOWER_OPENCL=0
+    continue
+  fi
+
+  if [ $ARG == "--disable-hsa" ]; then
+    LOWER_HSA=0
+    continue
+  fi
+ 
+
+
+  #####################################
+  # detect object or static library
+  #####################################
+
+  OBJS_TO_PROCESS=()
+
+
+  if [[ "$ARG" =~ [^[:space:]]+\.cpu$ ]]; then
+
+    cp $ARG $TEMP_DIR/kernel_cpu.o
+    NEW_LINK_CPU_ARG=$NEW_LINK_CPU_ARG" "$TEMP_DIR/kernel_cpu.o
+ 
+  elif [[ "$ARG" =~ [^[:space:]]+\.o$ ]]; then
+    # detected a .o file
+    if [ $VERBOSE == 2 ]; then
+      echo "detect object file to process further: $ARG"
+    fi
+
+    OBJS_TO_PROCESS+=( $ARG )
+  elif [[ "$ARG" =~ ^-l[^[:space:]]+$ ]] || [[ "$ARG" =~ [^[:space:]]+.a$ ]]; then
+
+    # proccess a static library
+
+    DETECTED_STATIC_LIBRARY=""
+
+    # detected whether it's an -l<library naem> option
+    if [[ "$ARG" =~ ^-l[^[:space:]]+$ ]]; then
+
+      # expand the option into a library name
+      STATIC_LIB_NAME="lib${ARG:2}.a"
+
+      if [ $VERBOSE == 2 ]; then
+        echo "looking for static library $STATIC_LIB_NAME"
+      fi
+
+      # look for the static library in the library search paths
+      for LIB_PATH in "${LIB_SEARCH_PATHS[@]}"
+      do
+        FULL_LIB_PATH=$LIB_PATH"/"$STATIC_LIB_NAME
+        if [ -f $FULL_LIB_PATH ]; then
+          if [ $VERBOSE == 2 ]; then
+            echo "$FULL_LIB_PATH detected"
+          fi
+          DETECTED_STATIC_LIBRARY=$FULL_LIB_PATH;
+          break;
+        fi
+      done
+    else
+      # this is .a static library file specified at the commad line
+      if [ -f $ARG ]; then
+        DETECTED_STATIC_LIBRARY=$ARG
+      fi
+    fi  # if [[ "$ARG" =~ ^-l[^[:space:]]+$ ]]; then
+
+    
+    KERNEL_UNDETECTED="1" 
+    if [[ $DETECTED_STATIC_LIBRARY != "" ]]; then
+      
+      # we found a static library library
+      if [ $VERBOSE == 2 ]; then
+        echo "processing static library $DETECTED_STATIC_LIBRARY"
+      fi
+
+      # detect whether the objects in the static library contain a .kernel section
+      KERNEL_UNDETECTED=`objdump -t $DETECTED_STATIC_LIBRARY | grep -q "\.kernel"; echo $?`
+      if [[ $KERNEL_UNDETECTED == "0" ]]; then
+
+        # .kernel section detected, extract the objects from the archieve
+
+        if [ $VERBOSE == 2 ]; then
+          echo "kernel detected in $DETECTED_STATIC_LIBRARY"
+        fi
+
+        CURRENT_DIR=$PWD
+        # extract the archive
+        FILE=`basename $DETECTED_STATIC_LIBRARY`
+        AR_TEMP_DIR=$TEMP_DIR"/"$FILE
+
+        if [ $VERBOSE == 2 ]; then
+          echo "creating temp dir: $AR_TEMP_DIR"
+        fi
+ 
+        mkdir -p $AR_TEMP_DIR
+        TEMP_AR_DIRS+=( $AR_TEMP_DIR )
+        cd $AR_TEMP_DIR
+        `ar x $DETECTED_STATIC_LIBRARY`
```

Comment:
If you see an error about libmcwamp.a when building the compiler, this patch https://github.com/RadeonOpenCompute/hcc/pull/9  will fix that